### PR TITLE
Allow project members to read ResourceQuota

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area usability
/kind enhancement
/priority normal

**What this PR does / why we need it**:
This PR enhances the `gardener.cloud:system:project-member-aggregation` and `gardener.cloud:system:project-viewer-aggregation` ClusterRoles by allowing read operations for ResourceQuotas.

**Release note**:

```improvement user
Project admins and viewers are now allowed to read `corev1.ResourceQuota` objects.
```

### CODE REVIEW

File: charts/gardener/controlplane/charts/application/templates/rbac-user.yaml [hunk 1]
```diff
@@ -180,6 +180,13 @@ rules:
   - get
   - list
   - watch
+- apiGroups:
+  - ""
+  resources:
+  - resourcequotas
+  verbs:
+  - get
+  - list
```
Comment (rfranzke) [inline charts/gardener/controlplane/charts/application/templates/rbac-user.yaml]:
```suggestion
  verbs:
  - get
  - list
  - watch
```

**Review by rfranzke — CHANGES_REQUESTED** [review]

**Comment by rfranzke** [comment]
> Project viewers are not permitted to read quotas, because I rather consider it an administrative task.

Why not? 👀 I don't get the difference.

**Comment by timuthy** [comment]
> Why not? 👀 I don't get the difference.

Principle of least privilege. As a viewer you cannot influence the quota in any way by creating/deleting resources, i.e. I don't see a use case for that audience at the moment, do you?

**Comment by rfranzke** [comment]
Hm, I cannot follow, sorry. As the name suggests, isn't the viewer role exactly used to "view" resources in the project namespace? The only exception are `Secret`s for obvious reasons. Viewers cannot influence shoots in any way either, still, they are allowed to see them. What's the difference with `ResourceQuota`s?

Despite from this theoretical point of view, I also see use-cases where end-users want to automate their own alerting or monitoring of such resource quotas. Why should they be forced to use a service account with `admin` role for that?

**Comment by timuthy** [comment]
> Despite from this theoretical point of view, I also see use-cases where end-users want to automate their own alerting or monitoring of such resource quotas. Why should they be forced to use a service account with admin role for that?

Makes sense, thanks. I'll add it.

**Review by danielfoehrKn — APPROVED** [review]

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**